crates/astria-bridge-withdrawer/src/bridge_withdrawer/startup.rs

```diff
@@ -0,0 +1,490 @@
+    }
+}
+
+/// Extracts the rollup height from the last transaction by the bridge account on the sequencer.
+/// Since all the withdrawals from a rollup block are batched into a single sequencer transaction,
+/// he rollup height can be extracted from the memo of any withdrawal action in the batch.
+///
+/// # Returns
+///
+/// The rollup height of the last batch of withdrawals.
+///
+/// # Errors
+///
+/// 1. The last transaction by the bridge account did not contain a withdrawal action.
+/// 2. The memo of the last transaction by the bridge account could not be parsed.
+/// 3. The block number in the memo of the last transaction by the bridge account could not be
+///   converted to a u64.
+fn rollup_height_from_signed_transaction(
+    signed_transaction: &SignedTransaction,
+) -> eyre::Result<u64> {
+    // find the last batch's rollup block height
+    let withdrawal_action = signed_transaction
+        .actions()
+        .iter()
+        .find(|action| matches!(action, Action::BridgeUnlock(_) | Action::Ics20Withdrawal(_)))
+        .ok_or_eyre("last transaction by the bridge account did not contain a withdrawal action")?;
+
+    let last_batch_rollup_height = match withdrawal_action {
+        Action::BridgeUnlock(action) => {
+            let memo: BridgeUnlockMemo = serde_json::from_slice(&action.memo)
+                .wrap_err("failed to parse memo from last transaction by the bridge account")?;
+            Some(memo.block_number.as_u64())
+        }
+        Action::Ics20Withdrawal(action) => {
+            let memo: Ics20WithdrawalFromRollupMemo = serde_json::from_str(&action.memo)
+                .wrap_err("failed to parse memo from last transaction by the bridge account")?;
+            Some(memo.block_number)
+        }
+        _ => None,
+    }
+    .expect("action is already checked to be either BridgeUnlock or Ics20Withdrawal");
+
+    info!(
+        last_batch.tx_hash = %telemetry::display::hex(&signed_transaction.sha256_of_proto_encoding()),
+        last_batch.rollup_height = last_batch_rollup_height,
+        "extracted rollup height from last batch of withdrawals",
+    );
+
+    Ok(last_batch_rollup_height)
+}
+
+#[instrument(skip_all)]
+async fn get_bridge_account_last_transaction_hash(
+    client: sequencer_client::HttpClient,
+    state: Arc<State>,
+    address: Address,
+) -> eyre::Result<BridgeAccountLastTxHashResponse> {
+    let res = tryhard::retry_fn(|| client.get_bridge_account_last_transaction_hash(address))
+        .with_config(make_sequencer_retry_config(
+            "attempt to fetch last bridge account's transaction hash from Sequencer; retrying \
+             after backoff"
+                .to_string(),
+        ))
+        .await
+        .wrap_err(
+            "failed to fetch last bridge account's transaction hash from Sequencer after a lot of \
+             attempts",
+        );
+
+    state.set_sequencer_connected(res.is_ok());
+
+    res
+}
+
+#[instrument(skip_all)]
+async fn get_tx(
```

Comment:
Can you give this a more verbose name? `get_sequencer_transaction_at_hash`? I mentioned it before, but this crate uses `tx` to mean:

1. a sending channel
2. an ethereum transaction
3. a sequencer (cometbft) transaction

Reply:
`get_tx` is the name of this in the tendermint client, but sure.